public/javascript/posts/BookAppointmentModal.js

```diff
@@ -149,7 +161,12 @@ angular.module('BookAppointmentApp')
       },
 
       paymentSelected: function(){
-        return true;
+        if(selected_event.prepay_amount > 0){
```

Comment:
As 'selected_event' is local variable in factory 'eventDetails', it can not be used as '$scope.selected_event'.